### ISSUE TITLE
Setting Buffer Length To Length of Current Audio Stream

### ISSUE THREAD

**Issue by patrickjquinn** [tumtumtum/StreamingKit#280]
Hi There,

You mention in your ReadMe that the buffer is variable length, yes you can set `initWithOptions` to use `.bufferSizeInSeconds = 300` in order to set the buffer large enough to accomidate most tracks without having to re-buffer or have a constant internet connection.  The problem is that the memory overhead of doing this is pretty significant (around 60 mb or so) and my application is already pretty resource intensive.

Is it possible to set the buffer size to be the length of the current stream?

**Comment by andrewebling** [comment]
I too would be interested in a way to do this.

**Comment by patrickjquinn** [comment]
I kind of thought that it was me doing something silly that was the reason i couldn't get this to work but is this actually a feature of the framework? Its a pretty minor gripe but would be really nice to have!

**Comment by tumtumtum** [comment]
At the moment the buffer is at the PCM (raw) audio level. For buffering compressed data the "easiest" thing to do would be to write a DataSourceWrapper that does the buffering of the incoming stream (HTTP, file, etc).

**Comment by patrickjquinn** [comment]
Good idea, so handle the stream in a manually created buffer which is set to the content size (reported by the http headers) and then feed it to the streaming kit instance as a source?

**Comment by tumtumtum** [comment]
Pretty much but you probably wanna start sending out data to STK before downloading the whole thing. If you're using the content size reported by the HTTP headers make sure you handle the case when there's no size (live stream)
